libc-bottom-half/cloudlibc/src/common/net.h

```diff
@@ -64,12 +66,18 @@ static inline int wasi_to_sockaddr(const struct __wasi_addr_port_t *restrict pee
     } else if (peer_addr->tag == __WASI_ADDRESS_FAMILY_INET6) {
       struct sockaddr_in6 addr6;
       addr6.sin6_family = AF_INET6;
-      addr6.sin6_flowinfo = 0;
-      addr6.sin6_scope_id = 0;
+      addr6.sin6_flowinfo = peer_addr->u.inet6.addr.flow_info1 << 16 & peer_addr->u.inet6.addr.flow_info0;
```

Comment:
I dubt this is correct.
It's probably an OR `| ` you want and not an AND `&` here.

Reply:
Also, better use some parenthesis to be safe,
```C
addr6.sin6_flowinfo = (peer_addr->u.inet6.addr.flow_info1 << 16 ) | (peer_addr->u.inet6.addr.flow_info0);
```

Reply:
You're absolutely right. Since we're not reading these on the wasmer side, I couldn't test the code.

Reply:
Fixed